src/platform/modulesmanager/src/ManagementModule.cpp

```diff
@@ -7,514 +7,476 @@
-{
-    return isValid;
-}
-
-bool ManagementModule::IsLoaded() const
-{
-    return mmiHandle ? true : false;
-}
 
-const std::vector<std::string> ManagementModule::GetSupportedComponents() const
-{
-    return info.components;
-}
-
-const std::vector<std::string> ManagementModule::GetReportedObjects(const std::string& componentName) const
-{
-    std::vector<std::string> objects;
-    if (reportedObjects.find(componentName) != reportedObjects.end())
+    // User Account
+    if (object.HasMember(g_mmiGetInfoUserAccount.c_str()))
     {
-        objects = reportedObjects.at(componentName);
-    }
-    return objects;
-}
-
-void ManagementModule::AddReportedObject(const std::string& component, const std::string& object)
-{
-    if (reportedObjects.find(component) == reportedObjects.end())
-    {
-        reportedObjects[component] = std::vector<std::string>();
-    }
-
-    if (std::find(reportedObjects[component].begin(), reportedObjects[component].end(), object) == reportedObjects[component].end())
-    {
-        reportedObjects[component].push_back(object);
-    }
-}
-
-bool ManagementModule::IsExportingMmi(const std::string path)
-{
-    void* handle = dlopen(path.c_str(), RTLD_LAZY);
-    if (nullptr == handle)
-    {
-        return false;
-    }
-
-    // Make sure all function exports are present
-    const std::vector<std::string> mmi_funcs = {MmiFuncMmiOpen, MmiFuncMmiClose, MmiFuncMmiSet, MmiFuncMmiGet, MmiFuncMmiGetInfo, MmiFuncMmiFree};
-    for (auto &f : mmi_funcs)
-    {
-        mmi_t funcPtr = (mmi_t)dlsym(handle, f.c_str());
-        if (nullptr == funcPtr)
+        if (object[g_mmiGetInfoUserAccount.c_str()].IsUint())
         {
-            OsConfigLogError(ModulesManagerLog::Get(), "Unable to call %s on %s", f.c_str(), path.c_str());
-            dlclose(handle);
-            return false;
+            info.userAccount = object[g_mmiGetInfoUserAccount.c_str()].GetUint();
+        }
+        else
+        {
+            OsConfigLogError(ModulesManagerLog::Get(), "Module info field '%s' is not an unsigned integer", g_mmiGetInfoUserAccount.c_str());
         }
     }
-    dlclose(handle);
 
-    return true;
+    return status;
 }
 
-int ManagementModule::CallMmiSet(const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes)
+MmiSession::MmiSession(std::shared_ptr<ManagementModule> module, const std::string& clientName, unsigned int maxPayloadSizeBytes) :
+    m_clientName(clientName),
+    m_maxPayloadSizeBytes(maxPayloadSizeBytes),
+    m_module(module),
+    m_handle(nullptr)
 {
-    int status = MMI_OK;
-
-    // Validate payload before calling MmiSet
-    if (IsValidMimObjectPayload(payload, payloadSizeBytes, ModulesManagerLog::Get()))
+    if (m_module)
     {
-        LoadModule();
-        status = mmiSet(mmiHandle, componentName, objectName, payload, payloadSizeBytes);
+        m_handle = m_module->CallMmiOpen(m_clientName.c_str(), m_maxPayloadSizeBytes);
```

Comment:
> m_handle = m_module->CallMmiOpen(m_clientName.c_str(), m_maxPayloadSizeBytes); [](http://example.com/codeflow?start=7&length=79)

Dangerous to invoke an external function such as MmiOpen from within constructor.

Leave constructors simple and move such initialization code to some Initialize() member function. Here and elsewhere.